tests/util/event_loop.cpp

```diff
@@ -204,6 +205,69 @@ void EventLoop::Impl::perform(std::function<void()> f)
     CFRunLoopWakeUp(m_loop.get());
 }
 
+#elif REALM_USE_ALOOPER
+
+static ALooper* s_main_looper;
+
+__attribute__((__constructor__))
+static void acquire_main_looper()
+{
+    s_main_looper = ALooper_prepare(0);
+    ALooper_acquire(s_main_looper);
+}
+
+__attribute__((__destructor__))
+static void release_main_looper()
+{
+    ALooper_release(s_main_looper);
+}
+
+bool EventLoop::has_implementation() { return true; }
+
+std::unique_ptr<EventLoop::Impl> EventLoop::Impl::main()
+{
+    return std::unique_ptr<Impl>(new Impl(s_main_looper));
```

Comment:
May as well use `std::make_unique` here.

Reply:
I tried that, but the `Impl` constructor is private.